build/squire-raw.js

```diff
@@ -3386,6 +3405,89 @@ var decreaseListLevel = function ( frag ) {
     return frag;
 };
 
+var getTextFromHTMLFragment = function ( self, frag ) {
+    var walker = getBlockWalker( frag ),
+        lines = [],
+        node;
+    while ( node = walker.nextNode() ) {
+        // Strip down to text only
+        lines.push( node.textContent );
+    }
+    return self._doc.createTextNode( lines.join( '\n' ) || '\n' );
+};
+
+var makePreformatted = function ( frag ) {
+    return this.createElement( 'PRE',
+        this._config.tagAttributes.pre, [
+            getTextFromHTMLFragment( this, frag )
+        ] );
+};
+
+var removePreformatted = function ( frag ) {
```

Comment:
I have to disagree that this echoes Squire#removeAllFormatting(). What I'm looking to do is maintain the line format as best it can be - and inside a PRE tag, whitespace linebreaks (`/\r?\n/`) are shown. So we'd want the content of the PRE element to be split by linebreaks, and turned into one defaultblock per line. As well, we need to split the PRE element at the range borders.

Both of those activities do have more graceful ways to do them than this, I'll grant. I'll make with finding them.